sync_diff_inspector/source/tidb.go

```diff
@@ -126,7 +127,33 @@ func (s *TiDBSource) GetCountAndMd5(ctx context.Context, tableRange *splitter.Ra
 	chunk := tableRange.GetChunk()
 
 	matchSource := getMatchSource(s.sourceTableMap, table)
-	count, checksum, err := utils.GetCountAndMd5Checksum(ctx, s.dbConn, matchSource.OriginSchema, matchSource.OriginTable, table.Info, chunk.Where, chunk.Args)
+	indexHint := ""
+	if s.sqlHint == "auto" && len(chunk.IndexColumnNames) > 0 {
+		// Since the index name is extracted from one data source,
+		// while another data source may have an index with same columns and different index name, which can pass the struct check,
+		// here we use the index column names to find the match index name again.
+		//
+		// For example:
+		//     Upstream:   idx1(c1, c2)
+		// 	   Downstream: idx2(c1, c2)
+		// Here we will store [c1, c2] from idx1 in the chunk, and use them to get idx2 as index hint.
+		if tableInfos, err := s.GetSourceStructInfo(ctx, tableRange.GetTableIndex()); err == nil {
+			for _, index := range dbutil.FindAllIndex(tableInfos[0]) {
+				if utils.IsIndexMatchingColumns(index, chunk.IndexColumnNames) {
+					indexHint = fmt.Sprintf("/*+ USE_INDEX(`%s`.`%s`, `%s`) */",
+						matchSource.OriginSchema,
+						matchSource.OriginTable,
+						index.Name.L,
+					)
```

Comment:
```suggestion
					indexHint = fmt.Sprintf("/*+ USE_INDEX(%s, %s) */",
						dbutil.TableName(matchSource.OriginSchema, matchSource.OriginTable),
						dbutil.ColumnName(index.Name.O),
					)
```

(yeah it is not a column but whatever)